course5/homework06/simple_string.cc

```diff
@@ -0,0 +1,188 @@
+#include "simple_string.h"
+#include <memory>
+#include <stdexcept>
+
+
+using namespace simple_string;
+
+String::String() = default;
+
+String::String(const char* initString)
+{
+    size = std::strlen(initString);
+    buffer = std::make_unique<char[]>(size + 1);
+    std::strcpy(buffer.get(), initString);
+}
+
+String::String(const String& src) // copy
+{
+    size = src.size;
+
+    if (src.buffer)
+    {
+        buffer = std::make_unique<char[]>(size + 1);
+        std::strcpy(buffer.get(), src.buffer.get());
+    }
+}
+
+String::String(String&& src) // move
+{
+    buffer = std::move(src.buffer);
+    size = src.size;
+
+    src.buffer = nullptr;
+    src.size = 0;
+}
+
```

Comment:
And this is a sing that you are using correct tools: not having to implement default constructor and destructor.